rotkehlchen/data_import/importers/_import_helpers.py

```diff
@@ -0,0 +1,241 @@
+from rotkehlchen.types import (
+    AssetAmount,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradeType,
+)
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def classify_transaction(
+    origin: str,
+    destination: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+    origin_amount: AssetAmount,
+    destination_amount: AssetAmount,
+    transaction_type: str,
+) -> typing.Callable:
+    if is_on_exchange(origin, destination):
+        if is_ledger_action(
+            origin_asset,
+            destination_asset,
+            origin_amount,
+            destination_amount,
+        ):
+            return TransactionType.LedgerAction
+        if is_trade(origin_asset, destination_asset):
+            return TransactionType.Trade
+        raise ValueError(
+            f'Could not classify transaction between {origin} and {destination} '
+            f'with type {transaction_type}',
+        )
+    if is_withdrawal(
+        transaction_type=transaction_type,
+        origin_asset=origin_asset,
+        destination_asset=destination_asset,
+    ) or is_deposit(
+        transaction_type=transaction_type,
+        origin_asset=origin_asset,
+        destination_asset=destination_asset,
+    ):
+        return TransactionType.AssetMovement
+    if is_trade(origin_asset=origin_asset, destination_asset=destination_asset):
+        return TransactionType.Trade
+    raise ValueError(
+        f'Could not classify transaction between {origin} and {destination} '
+        f'with type {transaction_type}',
+    )
+
+
+def check_transaction_type(
+    transaction_type: str,
+    origin: str,
+    destination: str,
+    target_platform: str,
+) -> None:
+    """Check if the transaction type is valid."""
+    if is_on_exchange(origin, destination):
+        return
+    if transaction_type == 'in' and origin != target_platform:
+        raise ValueError(
+            'Transaction Type and origin don\'t match. '
+            f'Transaction Type is {transaction_type} but origin is {origin}',
+        )
+    if transaction_type == "out" and destination != target_platform:
+        raise ValueError(
+            'Transaction Type and destination don\'t match. '
+            f'Transaction Type is {transaction_type} but destination is {destination}',
+        )
+
+
+def is_on_exchange(origin: str, destination: str) -> bool:
+    """Check if the transaction is on exchange."""
+    return origin == destination
+
+
+def is_same_asset(origin_asset: Asset, destination_asset: Asset) -> bool:
+    """Check if the transaction has the same origin and destination assets."""
+    return origin_asset == destination_asset
+
+
+def is_withdrawal(
+    transaction_type: str,
+    origin_asset: Asset,
+    destination_asset: Asset,
+) -> bool:
+    """Check if the transaction is a withdrawal."""
+    if not is_same_asset(origin_asset, destination_asset):
+        raise ValueError(
```

Comment:
We try to avoid raising `ValueError` as much as possible and use more specific errors instead. For example, we have `InputError`